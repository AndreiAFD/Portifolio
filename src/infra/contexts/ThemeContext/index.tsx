import React, { createContext, useState } from "react";
import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../core/theme/theme";

enum ThemeEnum {
  light = "light",
  dark = "dark",
}

interface AppThemeContextProps {
  theme?: string;
  handleChangeTheme?: (theme: ThemeEnum) => void;
}

export const AppThemeContext = createContext<AppThemeContextProps>({});

export const AppThemeContextProvider = ({ children }: { children: any }) => {
  const [actualTheme, setActualTheme] = useState<ThemeEnum>(ThemeEnum.light);

  const chakraTheme = extendTheme({
    colors: {
      ...theme[actualTheme],
    },
  });

  const handleChangeTheme = (nextTheme: ThemeEnum) => {
    setActualTheme(nextTheme);
  };

  return (
    <AppThemeContext.Provider
      value={{
        theme: actualTheme,
        handleChangeTheme,
      }}
    >
      <ThemeProvider theme={theme[actualTheme] as typeof theme["dark"]}>
        <ChakraProvider theme={chakraTheme}>{children}</ChakraProvider>
      </ThemeProvider>
    </AppThemeContext.Provider>
  );
};
