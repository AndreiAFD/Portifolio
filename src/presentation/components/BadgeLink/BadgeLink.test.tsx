import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { BadgeLink } from ".";
import { AppThemeContextProvider } from "../../../infra/contexts/ThemeContext";

const ComponentToTest = () => {
  return (
    <AppThemeContextProvider>
      <BadgeLink url="https://google.com" color="blue">
        Testing Google link
      </BadgeLink>
    </AppThemeContextProvider>
  );
};

describe("BadgeLink component", () => {
  it("should render correctly", () => {
    render(<ComponentToTest />);
    const badgeLink = screen.getByTestId("badgelink-container-testid");
    expect(badgeLink).toMatchSnapshot();
  });
  it("should navigate to url", () => {
    render(<ComponentToTest />);
    const link = screen.getByTestId("badgelink-testid");
    fireEvent.click(link);
  });
});
