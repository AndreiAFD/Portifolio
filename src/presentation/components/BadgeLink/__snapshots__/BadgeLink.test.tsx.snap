// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeLink component should render correctly 1`] = `
<button
  class="chakra-button css-1y6ftdq"
  data-testid="badgelink-container-testid"
  type="button"
>
  <a
    class="chakra-link css-3zcpgt"
    data-testid="badgelink-testid"
    href="https://google.com"
    target="_blank"
  >
    <img
      class="chakra-image css-19xcex3"
    />
    Testing Google link
  </a>
</button>
`;
